htsinfer/get_library_type.py

```diff
@@ -154,47 +165,68 @@ def _align_mates(self):
+                ):
+                    mate1.append(reads1.copy())
+                    reads1.clear()
+                if read1.reference_end:
+                    reads1.append(read1)
             previous_seq_id1 = seq_id1
         mate1.append(reads1.copy())
 
         read_counter = 0
         for read2 in samfile2:
-            seq_id2 = read2.query_name
-            if seq_id2 != previous_seq_id2 \
-                    and previous_seq_id2 is not None:
-                if self._compare_alignments(mate1[read_counter], reads2):
-                    concordant += 1
-                reads2.clear()
-                read_counter += 1
-            if read2.reference_end:
-                reads2.append(read2)
+            if not read2.flag & (1 << 2) and isinstance(read2.query_name, str):
+                seq_id2 = read2.query_name
+                if seq_id2 != previous_seq_id2 \
+                        and previous_seq_id2 is not None:
+                    if read_counter < len(mate1) and self._compare_alignments(
+                        mate1[read_counter], reads2
+                    ):
+                        concordant += 1
+                    reads2.clear()
+                    read_counter += 1
+                if read2.reference_end:
+                    reads2.append(read2)
             previous_seq_id2 = seq_id2
 
-        if self._compare_alignments(mate1[read_counter], reads2):
+        if read_counter < len(mate1) and self._compare_alignments(
+            mate1[read_counter], reads2
+        ):
             concordant += 1
+        LOGGER.debug(f"Number of aligned reads file 1: {len(mate1)}")
+        LOGGER.debug(f"Number of aligned reads file 2: {read_counter}")
+        LOGGER.debug(f"Number of concordant reads: {concordant}")
+        self._update_relationship(concordant, read_counter)
 
-        if (concordant / read_counter) >= self.cutoff:
+        samfile1.close()
+        samfile2.close()
+
+    def _update_relationship(self, concordant, read_counter):
+        """Helper function to update relationship based on alignment."""
+        try:
+            ratio = concordant / read_counter
+        except ZeroDivisionError:
             self.results.relationship = (
-                StatesTypeRelationship.split_mates
+                StatesTypeRelationship.not_available
             )
-            self.mapping.library_type.relationship \
-                = StatesTypeRelationship.split_mates
-            self.mapping.mapped = False
-            self.mapping.star_dirs = []
         else:
-            self.results.relationship = (
-                StatesTypeRelationship.not_mates
-            )
-
-        samfile1.close()
-        samfile2.close()
+            if ratio >= self.cutoff:
+                self.results.relationship = (
+                    StatesTypeRelationship.split_mates
+                )
+                self.mapping.library_type.relationship \
+                    = StatesTypeRelationship.split_mates
```

Comment:
I would use the `()` way of wrapping consistently (like in the line above)

Reply:
By the way, this is possible in Python (although here it will probably lead to a line that is too long):

```py
a = b = some_expression
```

i.e., you can assign the same value to multiple variables in the same statement.